finite-state/shared/src/main/scala/fs2/data/esp/ESP.scala

```diff
@@ -18,44 +18,47 @@ package fs2
 package data
 package esp
 
-import pattern._
-
 import cats.syntax.all._
+import cats.{Order, Show}
 
-import scala.collection.mutable.ListBuffer
 import scala.annotation.tailrec
+import scala.collection.mutable.ListBuffer
+
+import pattern._
 
 /** An Event Stream Processor is a generalization of the one defined in _Streamlining Functional XML Processing_.
   * It encodes its recognized patterns into a [[fs2.data.matching.DecisionTree Decision Tree]], including the state and depth. This flexibility allows for easily implementing
   * catch all rules, no matter what the state or depth is.
   */
-private[data] class ESP[F[_], Guard, InTag, OutTag](init: Int,
+private[data] class ESP[F[_], Guard, InTag, OutTag](val init: Int,
                                                     val params: Map[Int, Int],
                                                     val rules: DecisionTree[Guard, Tag[InTag], Rhs[OutTag]])(implicit
     F: RaiseThrowable[F]) {
 
-  def call[In, Out](env: Map[Int, Expr[Out]], q: Int, d: Int, args: List[Expr[Out]], in: Option[In])(implicit
+  def call[In, Out](env: Vector[Expr[Out]], q: Int, d: Int, args: List[Expr[Out]], in: Option[In], inline: Boolean)(
```

Comment:
Any specific reason to use `Vector`? Naively, I had expected the `env` to be rather small and we have `List` for `args`, so it looks/is inconsistent.